sdk/eventhub/Azure.Messaging.EventHubs/src/Consumer/EventHubConsumerClient.cs

```diff
@@ -768,7 +768,11 @@ public virtual async Task CloseAsync(CancellationToken cancellationToken = defau
         /// <returns>A task to be resolved on when the operation has completed.</returns>
         ///
         [SuppressMessage("Usage", "AZC0002:Ensure all service methods take an optional CancellationToken parameter.", Justification = "This signature must match the IAsyncDisposable interface.")]
-        public virtual async ValueTask DisposeAsync() => await CloseAsync().ConfigureAwait(false);
+        public virtual async ValueTask DisposeAsync()
+        {
+            await CloseAsync().ConfigureAwait(false);
+            GC.SuppressFinalize(this);
```

Comment:
My understanding has always been to avoid suppressing finalization unless you're explicitly using unmanaged resources.   However, this seems to no longer be considered best practice according to guidance. 

_([analysis rule reference](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1816))_